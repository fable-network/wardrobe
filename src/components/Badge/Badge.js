import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { pulseInfinite } from '../../animations';

const getBorderRadius = ({ size, radius, theme }) => {
  switch (radius) {
    case 'none':
      return '0';
    case 'small':
      return theme.borderRadiusSmall;
    case 'large':
    default:
      return size === 'large' ? '1.75em' : '1em';
  }
};

const getSize = ({ size, theme }) => {
  switch (size) {
    case 'small':
      return css`
        padding: 1px 0.75em 1px;
        font-size: ${theme.fontSizeXSmall};
        line-height: ${theme.lineHeightXSmall};
      `;
    case 'large':
      return `
        padding: calc(0.5em - 1px) 1.5rem;
        font-size: ${theme.fontSizeBase};
        line-height: ${theme.lineHeightBase};
      `;
    case 'normal':
    default:
      return css`
        padding: 1px 0.75em 1px;
        font-size: ${theme.fontSizeSmall};
        line-height: ${theme.lineHeightSmall};
      `;
  }
};

const cssPrimary = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.primary)};
  color: ${p => (p.inverse ? p.theme.primary : p.theme.white)};
  border-color: ${p => p.theme.primary};
`;

const cssSecondary = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.grey03)};
  color: ${p => (p.inverse ? p.theme.grey03 : p.theme.white)};
  border-color: ${p => p.theme.grey03};
`;

const cssSuccess = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.success)};
  color: ${p => (p.inverse ? p.theme.success : p.theme.white)};
  border-color: ${p => p.theme.success};
`;

const cssDanger = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.danger)};
  color: ${p => (p.inverse ? p.theme.danger : p.theme.white)};
  border-color: ${p => p.theme.danger};
`;

const cssWarning = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.warning)};
  color: ${p => (p.inverse ? p.theme.warning : p.theme.white)};
  border-color: ${p => p.theme.warning};
`;

const cssInfo = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.info)};
  color: ${p => (p.inverse ? p.theme.info : p.theme.white)};
  border-color: ${p => p.theme.info};
`;

const cssLight = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.grey03)};
  color: ${p => (p.inverse ? p.theme.grey03 : p.theme.grey01)};
  border-color: ${p => p.theme.grey03};
`;

const cssDark = css`
  background-color: ${p => (p.inverse ? p.theme.white : p.theme.grey01)};
  color: ${p => (p.inverse ? p.theme.grey01 : p.theme.white)};
  border-color: ${p => p.theme.grey01};
`;

const cssAnimated = css`
  animation: ${pulseInfinite} 3s infinite ease;
`;

const BadgeWrapper = styled('span')`
  display: inline-block;
  ${getSize};
  border-radius: ${getBorderRadius};
  text-align: center;
  color: ${p => p.theme.white};
  background-color: ${p => p.theme.primary};
  border: solid 1px;
  ${p => p.appearance === 'primary' && cssPrimary};
  ${p => p.appearance === 'secondary' && cssSecondary};
  ${p => p.appearance === 'success' && cssSuccess};
  ${p => p.appearance === 'danger' && cssDanger};
  ${p => p.appearance === 'warning' && cssWarning};
  ${p => p.appearance === 'info' && cssInfo};
  ${p => p.appearance === 'light' && cssLight};
  ${p => p.appearance === 'dark' && cssDark};
  ${p => p.animated && cssAnimated};
`;

const Badge = ({ appearance, children, animated, radius, size, inverse, ...otherProps }) => (
  <BadgeWrapper
    appearance={appearance}
    animated={animated}
    radius={radius}
    size={size}
    inverse={inverse}
    {...otherProps}
  >
    {children}
  </BadgeWrapper>
);

Badge.defaultProps = {
  appearance: 'primary',
  children: null,
  animated: false,
  radius: 'large',
  size: 'normal',
  inverse: false,
};

Badge.propTypes = {
  appearance: PropTypes.oneOf([
    'primary',
    'secondary',
    'success',
    'danger',
    'warning',
    'info',
    'light',
    'dark',
  ]),
  children: PropTypes.node,
  animated: PropTypes.bool,
  radius: PropTypes.oneOf(['large', 'small', 'none']),
  size: PropTypes.oneOf(['large', 'normal', 'small']),
  inverse: PropTypes.bool,
};

export default Badge;
