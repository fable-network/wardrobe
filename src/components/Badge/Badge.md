Badge default:

```jsx
<Badge appearance="primary">9</Badge>
```

Small Badge:

```jsx
<Badge appearance="primary" size="small">
  9
</Badge>
```

Large Badge, different color:

```jsx
<Badge appearance="danger" size="large">
  9
</Badge>
```

Badge with animation:

```jsx
<Badge appearance="success" animated>
  9
</Badge>
```

Badge with text:

```jsx
<div>
  <Badge>Hello world</Badge>
</div>
```

Large Badge with `radius="none"`:

```jsx
<Badge appearance="info" radius="none" size="large">
  Pre-pack
</Badge>
```

Inversed Badge with `radius="small"`:

```jsx
<Badge appearance="warning" radius="small" inverse>
  Acknowledged
</Badge>
```
