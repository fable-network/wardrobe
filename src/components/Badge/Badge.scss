//
// Badge
//
@import '../../globals/scss/colors';
@import '../../globals/scss/typography';
@import '../../globals/scss/variables';
@import '../../globals/scss/import-once';
@import '../../globals/scss/animations';

@include exports('badge') {
  .#{$prefix}--badge {
    display: inline-block;
    padding: 0.15em 0.8em;
    font-size: 80%;
    border-radius: 20px;
    text-align: center;
    color: #FFF;
    background-color: $color__skyBlue;

    &--primary {
      background-color: $color__skyBlue;
    }

    &--secondary {
      background-color: $color__stoneGrey;
    }

    &--success {
      background-color: $color__limeGreen;
    }

    &--danger {
      background-color: $color__flameRed;
    }

    &--warning {
      background-color: $color__apricotOrange;
    }

    &--light {
      color: $color__ravenBlack;
      background-color: $color__pearlWhite;
    }

    &--dark {
      background-color: $color__ravenBlack;
    }

    &--animated {
      animation: pulseInfinite 3s infinite ease;
    }
  }
}
