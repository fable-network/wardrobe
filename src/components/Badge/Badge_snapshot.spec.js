/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import Badge from './Badge';

describe('Badge Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      const wrapper1 = shallow(<Badge appearance="primary">9</Badge>);
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = shallow(
        <Badge appearance="primary" size="small">
          9
        </Badge>
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = shallow(
        <Badge appearance="danger" size="large">
          9
        </Badge>
      );
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = shallow(
        <Badge appearance="success" animated>
          9
        </Badge>
      );
      expect(wrapper4).toMatchSnapshot();

      const wrapper5 = shallow(
        <Badge appearance="info" radius="none" size="large">
          Pre-pack
        </Badge>
      );
      expect(wrapper5).toMatchSnapshot();

      const wrapper6 = shallow(
        <Badge appearance="warning" radius="small" inverse>
          Acknowledged
        </Badge>
      );
      expect(wrapper6).toMatchSnapshot();
    });
  });
});
