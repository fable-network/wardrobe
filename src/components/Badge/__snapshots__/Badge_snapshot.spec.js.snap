// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    animated={false}
    appearance="primary"
    inverse={false}
    radius="large"
    size="normal"
  >
    9
  </Badge>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animated": false,
      "appearance": "primary",
      "children": "9",
      "inverse": false,
      "radius": "large",
      "size": "normal",
    },
    "ref": null,
    "rendered": "9",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": false,
        "appearance": "primary",
        "children": "9",
        "inverse": false,
        "radius": "large",
        "size": "normal",
      },
      "ref": null,
      "rendered": "9",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Badge Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    animated={false}
    appearance="primary"
    inverse={false}
    radius="large"
    size="small"
  >
    9
  </Badge>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animated": false,
      "appearance": "primary",
      "children": "9",
      "inverse": false,
      "radius": "large",
      "size": "small",
    },
    "ref": null,
    "rendered": "9",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": false,
        "appearance": "primary",
        "children": "9",
        "inverse": false,
        "radius": "large",
        "size": "small",
      },
      "ref": null,
      "rendered": "9",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Badge Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    animated={false}
    appearance="danger"
    inverse={false}
    radius="large"
    size="large"
  >
    9
  </Badge>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animated": false,
      "appearance": "danger",
      "children": "9",
      "inverse": false,
      "radius": "large",
      "size": "large",
    },
    "ref": null,
    "rendered": "9",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": false,
        "appearance": "danger",
        "children": "9",
        "inverse": false,
        "radius": "large",
        "size": "large",
      },
      "ref": null,
      "rendered": "9",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Badge Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    animated={true}
    appearance="success"
    inverse={false}
    radius="large"
    size="normal"
  >
    9
  </Badge>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animated": true,
      "appearance": "success",
      "children": "9",
      "inverse": false,
      "radius": "large",
      "size": "normal",
    },
    "ref": null,
    "rendered": "9",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": true,
        "appearance": "success",
        "children": "9",
        "inverse": false,
        "radius": "large",
        "size": "normal",
      },
      "ref": null,
      "rendered": "9",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Badge Component snapshots should match snapshot 5`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    animated={false}
    appearance="info"
    inverse={false}
    radius="none"
    size="large"
  >
    Pre-pack
  </Badge>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animated": false,
      "appearance": "info",
      "children": "Pre-pack",
      "inverse": false,
      "radius": "none",
      "size": "large",
    },
    "ref": null,
    "rendered": "Pre-pack",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": false,
        "appearance": "info",
        "children": "Pre-pack",
        "inverse": false,
        "radius": "none",
        "size": "large",
      },
      "ref": null,
      "rendered": "Pre-pack",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Badge Component snapshots should match snapshot 6`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    animated={false}
    appearance="warning"
    inverse={true}
    radius="small"
    size="normal"
  >
    Acknowledged
  </Badge>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animated": false,
      "appearance": "warning",
      "children": "Acknowledged",
      "inverse": true,
      "radius": "small",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Acknowledged",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animated": false,
        "appearance": "warning",
        "children": "Acknowledged",
        "inverse": true,
        "radius": "small",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Acknowledged",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
